subprojects/file-watching/src/integTest/groovy/org/gradle/internal/watch/EnableFileSystemWatchingIntegrationTest.groovy

```diff
@@ -18,64 +18,117 @@ package org.gradle.internal.watch
     }
 
+    @Unroll
     def "can be enabled via #commandLineOption"() {
         buildFile << """
             apply plugin: "java"
         """
 
         when:
-        run("assemble", commandLineOption)
+        run("assemble", commandLineOption, "--info")
         then:
-        outputContains(INCUBATING_MESSAGE)
+        outputContains(expectedMessage)
 
         where:
-        commandLineOption << ["-D${StartParameterBuildOptions.WatchFileSystemOption.GRADLE_PROPERTY}=true", "--watch-fs"]
+        commandLineOption | expectedMessage
+        "--watch-fs"      | ENABLED_MESSAGE
+        "--no-watch-fs"   | DISABLED_MESSAGE
     }
 
-    def "deprecation message is shown when using the old property to enable watching the file system"() {
+    @Unroll
+    @SuppressWarnings('GrDeprecatedAPIUsage')
+    def "deprecation message is shown when using the old property to enable watching the file system (enabled: #enabled)"() {
         buildFile << """
             apply plugin: "java"
         """
         executer.expectDocumentedDeprecationWarning(
-                "Using the system property org.gradle.unsafe.vfs.retention to enable watching the file system has been deprecated. " +
-                        "This is scheduled to be removed in Gradle 7.0. " +
-                        "Use the gradle property org.gradle.unsafe.watch-fs instead. " +
-                        "See https://docs.gradle.org/current/userguide/gradle_daemon.html for more details."
+            "The org.gradle.unsafe.watch-fs system property has been deprecated. " +
+                "This is scheduled to be removed in Gradle 7.0. " +
+                "Please use the org.gradle.vfs.watch system property instead. " +
+                "See https://docs.gradle.org/current/userguide/gradle_daemon.html for more details."
         )
 
         expect:
-        succeeds("assemble", "-D${VirtualFileSystemServices.DEPRECATED_VFS_RETENTION_ENABLED_PROPERTY}=true")
+        succeeds("assemble", "-D${StartParameterBuildOptions.DeprecatedWatchFileSystemOption.GRADLE_PROPERTY}=${enabled}")
+
+        where:
+        enabled << [true, false]
+    }
+
+    @Unroll
+    @SuppressWarnings('GrDeprecatedAPIUsage')
+    def "deprecation message is shown when using old VFS retention property to enable watching the file system (enabled: #enabled)"() {
+        buildFile << """
+            apply plugin: "java"
+        """
+        executer.expectDocumentedDeprecationWarning(
+            "Using the system property org.gradle.unsafe.vfs.retention to enable watching the file system has been deprecated. " +
+                "This is scheduled to be removed in Gradle 7.0. " +
+                "Use the gradle property org.gradle.vfs.watch instead. " +
+                "See https://docs.gradle.org/current/userguide/gradle_daemon.html for more details."
+        )
+
+        expect:
+        succeeds("assemble", "-D${VirtualFileSystemServices.DEPRECATED_VFS_RETENTION_ENABLED_PROPERTY}=${enabled}")
```

Comment:
Shall we remove the `DEPRECATED_VFS_RETENTION_ENABLED_PROPERTY` directly?

Reply:
That would be extra work at this point. I'd say we remove it with the other deprecations in 7.0.

Reply:
Fine, let's leave it till 7.0.